### DISCUSSION TITLE
Can a listening socket initiate connection? If so, does the socket not listen anymore?

### DISCUSSION THREAD

**Discussion by livelykitten** [ANLAB-KAIST/KENSv3#161] [Project Q&A 2. Transport Layer (TCP)] [answered]
The state diagram allows transition from LISTEN to SYN_SENT
does that mean we have to implement such cases?

**Comment by Pusnow** [comment, marked as answer]
> Although the transition from LISTEN to SYN_SENT is allowed by TCP, there is no way to do this using the sockets API (i.e., a connect is not allowed after a listen).
> *TCP IP Illustrated Vol. 2. p.827*

So you don't need to implement the transition in KENS.

**Reply by livelykitten** [reply]
thank you
